cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-upload-sync.php

```diff
@@ -299,4 +306,48 @@ private function update_content( $attachment_id ) {
 			}
 		}
 	}
+
+	public function add_suffix_maybe( $attachment_id, $suffix = null ) {
+
+		// Test if asset exists by calling just the head on the asset url, to prevent API rate limits.
+		$options   = $this->media->get_upload_options( $attachment_id );
+		$public_id = $options['public_id'];
+
+		$url         = $this->connect->api->cloudinary_url( $public_id . $suffix, array( 'transformation' => array( array( 'fetch_format' => 'auto' ) ) ) );
+		$req         = wp_remote_head( $url, array( 'body' => array( 'rdm' => wp_rand( 100, 999 ) ) ) );
+		$asset_error = strtolower( wp_remote_retrieve_header( $req, 'x-cld-error' ) );
+		$code        = wp_remote_retrieve_response_code( $req );
+
+		// If the request is not a 404 & does not have a cld-error header stating resource not found, it exists and should be checked that it's not a resync or generate a prefixed ID.
+		if ( 404 !== $code && false === strpos( $asset_error, 'resource not found' ) ) {
+
+			// Get the attachment type.
+			$type      = $this->media->get_media_type( $attachment_id );
+			$cld_asset = $this->plugin->components['connect']->api->get_asset_details( $public_id . $suffix, $type );
+			if ( ! is_wp_error( $cld_asset ) && ! empty( $cld_asset['public_id'] ) ) {
+				$context_guid    = null;
+				$attachment_guid = md5( get_the_guid( $attachment_id ) );
+
+				// Exists, check to see if this asset originally belongs to this ID.
+				if ( ! empty( $cld_asset['context'] ) && ! empty( $cld_asset['context']['custom'] ) && ! empty( $cld_asset['context']['custom']['guid'] ) ) {
+					$context_guid = $cld_asset['context']['custom']['guid'];
+				}
+
+				// Generate new ID only if context ID is not related.
+				if ( $context_guid !== $attachment_guid ) {
+					// Generate a new ID with a uniqueID prefix.
+					$suffix = '-' . uniqid( $attachment_id );
+
+					// Return new potential suffixed ID.
+					return $this->add_suffix_maybe( $attachment_id, $suffix );
+				}
+			}
+		}
+		// Save defined suffix.
+		$this->media->update_post_meta( $attachment_id, Sync::META_KEYS['suffix'], $suffix );
```

Comment:
@DavidCramer 

What do you think of making the suffix already part of the `public_id` key? This would remove the need to append it across the source code.
We can eventually store the original in a `public_id_raw` key for cases when we need to find a new suffix.
This way we wouldn't even need to store the `suffix` key.

What do you think?

Reply:
@pereirinha - went back and forth with this one. But in the end, the. fact that people can filter the ID, I want to ensure they don't try to change the location. So, the folder it technically separate, but still stored.